test/test_build.py

```diff
@@ -170,15 +181,20 @@ def del_ami():
     journal_log_path.write_text(journal_output, encoding="utf8")
     bib_output_path.write_text(bib_output, encoding="utf8")
 
-    yield ImageBuildResult(
-        image_type, generated_img, target_arch, username, password,
-        bib_output, journal_output, metadata)
+    results = []
+    for image_type in image_types:
```

Comment:
I'm slightly uneasy that most of the existing tests now get a list which makes everything a bit more verbose. I wonder if we could just extract most of image_type_fixture into a normal python helper and then just call that helper in a dedicated test (because AIUI we cannot re-use a cached image, right?). And maybe in this test we export all the targets we support ? wdyt?

Reply:
[..]
> > (because AIUI we cannot re-use a cached image, right?)
> 
> Not sure what you mean. Why not?

Sorry, what I mean is right now if we say "qcow2" in the fixture and we have it already we never run bib (which makes sense). But for the qcow2+raw case we will not be able to reuse the qcow2 even if it exists. I think this is already encoded in the tests so I'm probably not adding anything new here :)

Reply:
This is actually something I'm wondering about.  It's probably trivial but now I'm thinking what if we test one scenario and then break the other.  We can test the multibuild and the single build scenarios by having two cases, `raw+qcow2` and `ami`, but what if we refactor something and somehow break the single `qcow2` case?  We'd never catch it :/